cli/flox-rust-sdk/src/models/manifest/composite/mod.rs

```diff
@@ -99,10 +89,16 @@ struct CompositeManifest {
     deps: Vec<(String, Manifest)>,
 }
 
+#[derive(Clone, Debug)]
+#[enum_dispatch(ManifestMergeTrait)]
+pub(crate) enum ManifestMerger {
```

Comment:
Didn’t get a chance to review this yesterday, but why do we have an enum with one variant and use enum dispatch on it?

Reply:
Well I don't think we need the trait at this point (YAGNI), but I think we added it so we could have the flexibility of adding another merge strategy. If we want that flexibility, the locking code that uses the merger needs to use a trait object or an enum like this. And I think we should use an enum for consistency with what we do everywhere else.